internal/compiler/generator/rust.rs

```diff
@@ -1015,6 +1031,15 @@ fn generate_sub_component(
                 #subtree_index_function
             }
 
+            fn item_geometry(self: ::core::pin::Pin<&Self>, index: u32) -> sp::LogicalRect {
+                #![allow(unused)]
+                let _self = self;
+                match index {
+                    #(#item_geometry_branch)*
+                    _ => ::core::default::Default::default()
+                }
```

Comment:
```suggestion
                sp::euclid::rect(match index {
                    #(#item_geometry_branch)*
                    _ => return ::core::default::Default::default();
                })
```

Reply:
That doesn't work because some of the branches in item_geometry_branch are calling `item_geometry()` functions in sub components

Reply:
ah, maybe with `return` it will work

Reply:
Ok, last commit does it now